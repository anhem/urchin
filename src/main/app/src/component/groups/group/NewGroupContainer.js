import React, {Component} from 'react'
import {connect} from 'react-redux'
import NewGroup from './NewGroup';
import {createGroup, setGroup} from '../../../action/groupAction';

class NewGroupContainer extends Component {

    render() {
        return (
            <NewGroup
                group={this.props.group}
                fieldErrors={this.props.fieldErrors}
                callbacks={{
                    setGroup: this.props.setGroup,
                    createGroup: this.props.createGroup,
                }}
            />
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setGroup: (group) => {
            dispatch(setGroup(group))
        },
        createGroup: (group) => {
            dispatch(createGroup(group))
        }
    }
};

const mapStateToProps = (state) => {
    return {
        group: state.groupData.group || {},
        fieldErrors: state.groupData.fieldErrors || {}
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(NewGroupContainer)