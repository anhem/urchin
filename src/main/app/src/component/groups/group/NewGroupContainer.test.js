import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {NewGroupContainer} from "./NewGroupContainer";

describe('NewGroupContainer', () => {

    let props = {
        group: {},
    };

    it('renders without crashing', () => {

        expect(mount(<NewGroupContainer {...props}/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<NewGroupContainer {...props}/>))).toMatchSnapshot();
    });
});
