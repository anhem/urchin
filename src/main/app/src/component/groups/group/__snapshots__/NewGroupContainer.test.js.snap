// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGroupContainer match snapshot 1`] = `
<NewGroup
  callbacks={
    Object {
      "createGroup": undefined,
      "setGroup": undefined,
    }
  }
  group={Object {}}
/>
`;
